File: project/core/module/rest-service/src/main/java/vdi/service/rest/conversion/SampleYearRange.java
package vdi.service.rest.conversion;

import org.jetbrains.annotations.NotNull;
import vdi.service.rest.generated.model.SampleYearRangeImpl;

public class SampleYearRange extends SampleYearRangeImpl {
  public SampleYearRange(@NotNull vdi.model.meta.SampleYearRange range) {
    setStart(range.getStart());
    setEnd(range.getEnd());
  }
}

File: project/core/module/rest-service/src/main/kotlin/vdi/service/rest/server/conversion/SampleYearRange.kt
package vdi.service.rest.server.conversion

import vdi.service.rest.generated.model.SampleYearRangeImpl
import vdi.model.meta.SampleYearRange as InternalYearRange
import vdi.service.rest.generated.model.SampleYearRange as RamlYearRange

internal fun SampleYearRange(range: InternalYearRange): RamlYearRange =
  SampleYearRangeImpl().apply {
    start = range.start
    end = range.end
  }